### PR TITLE
[feature](function) add approx_top_sum aggregation function

### PR DESCRIPTION
**Pull request by zzzxl1993**
### What problem does this PR solve?

Problem Summary:

1. The function approx_top_sum has been implemented. Here is an example of its usage: select approx_top_sum(c1, c2, c3, 10, 300) from tbl.

### Release note

None

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [ ] No.
    - [x] Yes. 

### Check List (For Reviewer who merge this PR)

- [x] Confirm the release note
- [x] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

File: regression-test/suites/inverted_index_p0/test_index_approx_top_k.groovy [hunk 1]
```diff
@@ -88,15 +88,11 @@ suite("test_index_approx_top_k", "p0"){
 
         sql """ set debug_skip_fold_constant = true; """
         qt_sql """ select clientip, count(*) as count from ${tableName} group by clientip order by count desc, clientip asc limit 10; """
-        qt_sql """ select approx_top_k(clientip) from ${tableName}; """
-        qt_sql """ select approx_top_k(clientip, 10) from ${tableName}; """
```
Comment (superdiaodiao) [inline regression-test/suites/inverted_index_p0/test_index_approx_top_k.groovy:92]: May I ask why deleting these tests?

Comment (zzzxl1993) [reply regression-test/suites/inverted_index_p0/test_index_approx_top_k.groovy:92]: Currently, it is difficult to implement default values for the variable arguments in the function framework.  It is hoped that users will directly specify the last two parameters.

**Comment by zzzxl1993** [comment]
run buildall

File: be/src/vec/aggregate_functions/aggregate_function.h [hunk 2]
```diff
@@ -43,7 +43,7 @@ class IDataType;
 
 struct AggregateFunctionAttr {
     bool enable_decimal256 {false};
-    std::vector<std::pair<std::string, bool>> column_infos;
+    std::vector<std::string> column_names;
```
Comment (xiaokang) [inline be/src/vec/aggregate_functions/aggregate_function.h:46]: Why change this basic data structure?

File: be/src/vec/aggregate_functions/aggregate_function_approx_top_sum.h [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+            rapidjson::Writer<rapidjson::StringBuffer> sub_writer(sub_buffer);
+            sub_writer.StartObject();
+            const char* begin = result.key.data;
+            for (size_t i = 0; i < _column_names.size(); i++) {
+                begin = argument_columns[i]->deserialize_and_insert_from_arena(begin);
+                std::string row_str = argument_types[i]->to_string(*argument_columns[i], 0);
+                sub_writer.Key(_column_names[i].data(), _column_names[i].size());
+                sub_writer.String(row_str.data(), row_str.size());
+            }
+            sub_writer.Key("count");
```
Comment (xiaokang) [inline be/src/vec/aggregate_functions/aggregate_function_approx_top_sum.h:224]: sum

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/ApproxTopK.java [hunk 4]
```diff
@@ -55,9 +55,19 @@ public ApproxTopK(boolean distinct, boolean alwaysNullable, Expression... varArg
 
     @Override
     public void checkLegalityBeforeTypeCoercion() {
-        if (arity() < 1) {
+        if (arity() < 3) {
```
Comment (xiaokang) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/ApproxTopK.java:58]: Why change arguments num from 1 to 3?

**Comment by zzzxl1993** [comment]
run buildall

File: be/src/vec/aggregate_functions/aggregate_function_approx_top_sum.h (hunk 3 above)
Comment (zzzxl1993) [reply be/src/vec/aggregate_functions/aggregate_function_approx_top_sum.h]: done

File: be/src/vec/aggregate_functions/aggregate_function.h (hunk 2 above)
Comment (zzzxl1993) [reply be/src/vec/aggregate_functions/aggregate_function.h:46]: Optimize the redundant data structures added in approx_top_k.

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/ApproxTopK.java (hunk 4 above)
Comment (zzzxl1993) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/ApproxTopK.java:58]: The existing function framework code cannot implement approx_top_k and approx_top_sum with default values.

**Comment by zzzxl1993** [comment]
run buildall

**Review by xiaokang — APPROVED** [review]
LGTM

**Review by qidaye — APPROVED** [review]
LGTM

**Pull request merged**